### PR TITLE
RD-3010 - support dynamodb streams

### PR DESCRIPTION
**Pull request by saartochner**

### CODE REVIEW

File: src/main/java/io/lumigo/core/utils/AwsUtils.java [hunk 1]
```diff
@@ -47,6 +50,24 @@ public static TriggeredBy extractTriggeredByFromEvent(Object event) {
                         && ((DynamodbEvent) event).getRecords().size() > 0) {
                     triggeredBy.setArn(
                             ((DynamodbEvent) event).getRecords().get(0).getEventSourceARN());
+                    DynamodbEvent.DynamodbStreamRecord firstRecord =
+                            ((DynamodbEvent) event).getRecords().get(0);
+                    triggeredBy.setArn(firstRecord.getEventSourceARN());
```
Comment (uri-p) [inline src/main/java/io/lumigo/core/utils/AwsUtils.java:55]: This set is already called before, remove one of the calls

File: src/main/java/io/lumigo/core/utils/AwsUtils.java [hunk 2]
```diff
@@ -47,6 +50,24 @@ public static TriggeredBy extractTriggeredByFromEvent(Object event) {
...
+                        triggeredBy.setApproxEventCreationTime(
+                                firstRecord
+                                        .getDynamodb()
+                                        .getApproximateCreationDateTime()
+                                        .getTime());
+                    }
+                    List<String> messageIds =
+                            ((DynamodbEvent) event)
+                                    .getRecords().stream()
+                                            .map(AwsUtils::extractMessageIdFromDynamodbRecord)
```
Comment (uri-p) [inline src/main/java/io/lumigo/core/utils/AwsUtils.java:67]: How many values can return here? maybe we need a hard limit

File: src/main/java/io/lumigo/core/utils/StringUtils.java [hunk 3]
```diff
@@ -54,4 +59,24 @@ public static String extractStringForStream(InputStream inputStream, int size) {
...
+
+    private static String bytesToHex(byte[] bytes) {
+        StringBuilder sb = new StringBuilder();
+        for (byte b : bytes) {
+            sb.append(String.format("%02x", b));
+        }
+        return sb.toString();
+    }
+
+    public static String buildMd5Hash(Map<String, AttributeValue> item) {
```
Comment (uri-p) [inline src/main/java/io/lumigo/core/utils/StringUtils.java:71]: I think the API should not get AttributeValue which is AWS property.
Looks like we need 2 modify it to string outside of this function

Comment (uri-p) [reply src/main/java/io/lumigo/core/utils/StringUtils.java:71]: The api will be a string to md5

File: src/main/java/io/lumigo/core/utils/AwsUtils.java (hunk 2 above)
Comment (saartochner) [reply src/main/java/io/lumigo/core/utils/AwsUtils.java:67]: It's configurable by the user, and the default is 50.
But you're right, I'll limit it with 50 just to not get exploded :)

**Pull request merged**